core/src/models/Property.ts

```diff
@@ -331,13 +331,17 @@ export class Property extends LoggedModel<Property> {
       : CachedProperties.properties;
   }
 
-  static async findOneWithCache(value: string, modelId?: string, key = "id") {
+  static async findOneWithCache(
+    value: string,
+    modelId?: string,
+    lookupKey: keyof Property = "id"
+  ) {
     const properties = await Property.findAllWithCache(modelId);
-    let property = properties.find((p) => p.key === value);
+    let property = properties.find((p) => p[lookupKey] === value);
 
     if (!property) {
       property = await Property.findOne({
-        where: { [key]: value },
+        where: { [lookupKey]: value },
```

Comment:
This bug was leading to a /ton/ of SQL selects on every usage of `record.getProprties()` -> `recordProperty.getValue()` -> `recordProperty.ensureProperty()` -> `Property.findOneWithCache()`.

Prior to this fix, we were never finding a match because p.key (literal) was never matching an id, which is what "key" should have been.  This wasn't caught in tests, because we disable the cache in tests :/

Reply:
Any idea how long this bug has been out in the wild?

Reply:
@pedroslopez any suggestions on how we could test this one?

Reply:
@andyjih probably for at least a few months

Reply:
I think we should definitely check that we're returning the cached properties in the tests. One idea is setting `CachedProperties.expires` explicitly to something in the future, and setting something like `_isCached` on the cached property within the test that we can check on the returned property from `findOneWithCache`.

Here's a quick example as an update to https://github.com/grouparoo/grouparoo/blob/463d74aaf43e6d9635efea9c9dd333cf396f11a4/core/__tests__/models/property/localCache.ts#L112

```ts
CachedProperties.expires = new Date().getTime() + 5000;
const cachedEmail = CachedProperties.properties.find(
  (p) => p.id === emailProperty.id
);
// @ts-ignore
cachedEmail.__isCached = true;
const found = await Property.findOneWithCache(emailProperty.id);
expect(found.key).toBe("email");
// @ts-ignore
expect(found.__isCached).toBe(true);
```

Typescript yells at you because `__isCached` isn't in the type, but I think it's _ok_ for the test.

Now, one thing I did notice is that in the tests the `id` and `key` are both the same (`email`), so the test would still pass in this case. They should be different to actually test this properly.

Reply:
Nice suggestion! messing with the cached object to ensure we get it back is a good path forward

Reply:
better cache tests via 09d6e77dc9e18b0c5638ee9d22c00a8640643ddf